scheduler/src/cook/mesos/api.clj

```diff
@@ -1331,6 +1331,36 @@
     :post! (fn [ctx]
              (apply set-limit-fn conn (get-in ctx [:request :body-params :user]) (reduce into [] (::limits ctx))))}))
 
+
+;; /failure-reasons
+
+(s/defschema FailureReasonsResponse
+  [{:code s/Int
+    :name s/Str
+    :description s/Str
+    :mea_culpa s/Bool
+    (s/optional-key :failure_limit) s/Int}])
+
+(defn reason-entity->consumable-map
+  [default-failure-limit e]
+  (cond-> {:code (:reason/code e)
+           :name (name (:reason/name e))
+           :description (:reason/string e)
+           :mea_culpa (:reason/mea-culpa? e)}
+    (:reason/mea-culpa? e)
+    (assoc :failure_limit (or (:reason/failure-limit e) default-failure-limit))))
```

Comment:
Not sure it's a great match...   of the 6 keys in the schema, only 2 are transformed directly in the manner that (remove-datomic-namespacing) would do it (simply removing the namespace).  I'm not including "mesos-reason" at all (don't think it's relevant to consumers of this API), I'm renaming "string" to "description" (more consumer friendly name for this attribute), and I'm converting "me-culpa?" to "mea_culpa" and "failure-limit" to "failure_limit" (these are more normal for JSON api's)